services/web-server/src/login/strategies/github.js

```diff
@@ -46,11 +46,35 @@ module.exports = class Github {
     user.identity = `${this.identityProviderId}/${userId}|${encode(username)}`;
 
     // take a user and attach roles to it
-    // this.addRoles(userProfile, user);
+    await this.addRoles(username, userId, user);
 
     return user;
   }
 
+  async addRoles(username, userId, user) {
+    const [orgsErr, orgs] = await tryCatch(this.githubClient.orgsFromUsername(username));
+
+    if (orgsErr) {
+      throw orgsErr;
+    }
+
+    const roles = [].concat(
+      ...(
+        await Promise.all(orgs.map(async ({ login: org }) => {
+          const [reposErr, repos] = await tryCatch(this.githubClient.reposFromOrg(org));
+
+          if (reposErr) {
+            throw reposErr;
+          }
+
+          return repos.map(repo => `github-group:${org}/${repo.name}`);
```

Comment:
If I understand correctly, this will end up issuing roles for every repo in any organization of which the user is a public member.  There are a few issues with that:
 * It's only public org membership
 * It will create scopes for repos to which the user doesn't have write access
 * Many users will be in orgs with hundreds of repos, so the resulting credentials will have hundreds of roles and not fit in an HTTP header

Reply:
> It's only public org membership

Good observation. We should look into adding private ones as well.

> It will create scopes for repos to which the user doesn't have write access

We should fix that :)

> * Many users will be in orgs with hundreds of repos, so the resulting credentials will have hundreds of roles and not fit in an HTTP header

The size of the request is currently set to 128kb but I think we could just increase that to something like 256kb. Let me know if you think that's not the right approach.

https://github.com/taskcluster/taskcluster/blob/b34aacc27c4e96565b598abf7c638e8fa3ce5530/ui/web-ui-nginx-site.conf#L32

Reply:
That nginx config is only for the UI, which only handles GET requests for UI assets.  And https://stackoverflow.com/questions/3326210/can-http-headers-be-too-big-for-browsers suggests Firefox's limit is 10K.

Reply:
You're right. I tested this locally and found that FF currently has it at around 12kb.